### ISSUE TITLE
What does the `lr` parameter in `.fit` do? Should I change it? 

### ISSUE THREAD

**Issue by wariobrega** [cantinilab/scconfluence#3]
Hey,

I just wanted to ask for a short explanation for the `lr` parameter of the .fit( method, that controls the learning rate. 
 
Does changing the default value of `0.003` significantly impact the embedding generated by scConfluence?

Have you played around with it?

Do you have any general tips on how to play around with it?

Thanks (and congratulations again for the method!)

**Comment by jules-samaran** [comment]
Hi and thank you for raising this issue!

The learning rate (or `lr`) parameter determines the size of the step taken at each iteration of the gradient descent algorithm used to train the parameters of scConfluence. While a very small learning rate can slow down the learning significantly, using a very large learning rate can lead to "jumping over" interesting loss minima and thus a worst performance. 

While developing scConfluence, I did a small grid search on the value of the learning rate and found the proposed default value of `0.003` to offer a good trade-off between performance and speed of convergence. I didn't observe any gain from the use of popular `pytorch` "lr schedulers" and therefore refrained from including them. If you were to try to further tune the `lr` for your application, I'd recommend trying smaller values than the default one since scConfluence's training generally converges already quite fast.

Take care,
